app/src/main/java/com/epfl/drawyourpath/path/cache/RunEntity.kt

```diff
@@ -39,7 +45,7 @@ data class RunEntity(
         fun fromRunsToEntities(userId: String, runs: List<Run>, sync: Boolean = true): List<Pair<RunEntity, List<PointsEntity>>> {
             return runs.map { run ->
                 Pair(
-                    RunEntity(userId, run.getStartTime(), run.getEndTime(), run.getDuration(), sync),
+                    RunEntity(userId, run.getStartTime(), run.getDuration(), run.getEndTime(), run.predictedShape, run.similarityScore, sync),
                     fromPathToEntity(userId, run.getStartTime(), run.getPath()),

```

Comment:
For function with a lot of arguments : you can write the arguments call like that: 
RunEntity(userId = userId, startTime = run.getStartTime().....)
It will be more readable and evict some errors.